docs/running_lbann.rst

```diff
@@ -1,96 +1,434 @@
+    (compute nodes on a system are typically subdivided into multiple
+    groups, e.g. for batch jobs and for debugging).
+
+  - Familiarize yourself with the rules for the systems you use
+    (e.g. the expected work for each partition, time limits, job
+    submission limits) and be a good neighbor.
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Model components
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note:: `A major refactor of core model infrastructure
+          <https://github.com/LLNL/lbann/pull/916>`_ is pending. This
+          documentation will be updated once it is merged and the
+          interface stabilized.
+
++ Layer: A tensor operation, arranged within a directed acyclic graph.
+
+  - During evaluation ("forward prop"), a layer receives input tensors
+    from its parents and sends an output tensor to each child.
+
+  - During automatic differentiation ("backprop"), a layer receives
+    "input error signals" (objective function gradients w.r.t. output
+    tensors) from its children and sends "output error signals"
+    (objective function gradients w.r.t. input tensors) to its
+    parents. If the layer has any associated weights, it will also
+    compute objective function gradients w.r.t. the weights.
+
+  - Most layers require a specific number of parents and children, but
+    LBANN will insert layers into the graph if there is a mismatch and
+    the intention is obvious. For example, if a layer expects one
+    child but has multiple, then a split layer (with multiple output
+    tensors all identical to the input tensor) is inserted. Similarly,
+    if a layer has fewer children than expected, dummy layers will be
+    inserted. However, this does not work if there is any
+    ambiguity. In such cases (common with input and slice layers), it
+    is recommended to manually insert identity layers so that the
+    parent/child relationships are absolutely unambiguous.
+
++ Weights: A tensor consisting of trainable parameters, typically
+  associated with one or more layers. A weights owns an initializer to
+  initially populate its values and an optimizer to find values that
+  minimize the objective function.
+
+  - A weights without a specified initializer will use a zero
+    initializer.
+
+  - A weights without a specified optimizer will use the model's
+    default optimizer.
+
+  - If a layer requires weightses and none are specified, it will
+    create the needed weightses. The layer will pick sensible
+    initializers and optimizers for the weightses.
+
+  - The dimensions of a weights is determined by their associated
+    layers. The user can not set it directly.
+
+.. note:: It is unfortunate that the deep learning community has
+   settled upon the plural word "weights" to describe tensors of
+   trainable parameters. Rather than using awkward and ambiguous
+   phrases like "set of weights," we have given up on grammar and
+   refer to "weights" (singular) and "weightses" (plural).
```

Comment:
Gollum runs the LBANN terminology team, I see.

Reply:
💍

Reply:
This isn't a very good solution, although I couldn't think of anything better.

Reply:
I've reduced the Gollum-talk by using the term "weight tensor" everywhere. It's a little awkward and doesn't get rid of everything (we talk about `Weights`es in the Python section), but it's a step forward.